core/lib/src/lib.rs

```diff
@@ -98,7 +98,7 @@
 //! module documentation](/rocket/local) and the [testing chapter of the
 //! guide](https://rocket.rs/guide/testing/#testing) include detailed examples.
 
-#[allow(unused_imports)] #[macro_use] extern crate rocket_codegen_next;
+#![allow(unused_imports)] #[macro_use] extern crate rocket_codegen_next;
```

Comment:
You're right -- that makes it an inner attribute (applying to the file as a whole) instead of an outer attribute (applying to the next item).

The Clippy documentation for that warning implied that it believed that the attribute was not being applied to anything, because it wasn't the "outer" part of anything else. I'll fix.

Reply:
Specifically -- https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#useless_attribute, "Lint attributes have no effect on crate imports." So probably the thing to do is just remove the `allow()`.

Reply:
Filed https://github.com/rust-lang-nursery/rust-clippy/issues/2932